### PR TITLE
[CI] Adopt the 1ES template

### PR DESCRIPTION
**Pull request by mandel-macaque**
There are a number of changes that were needed:

1. Use 1ES approved images. We moved all the windows-latests images to a 1ES image and moved all ubuntu ones to windows.
2. Update the tasks that do the uploads. 1ES provides a new task and will fail if other one is used.
3. Removed the workspace clean directive. Is not really needed and it confused the template.
4. Configured all the governance checks to be run by the template. Removed our own implementation of those tasks except for APIScan which is not included in the template.

Setting to draft since we need our macOS pools to be approved else the 1ES template test will fail and not run the jobs.

### CODE REVIEW

File: tools/devops/automation/templates/build/build-pkgs.yml [hunk 1]
```diff
@@ -77,11 +77,11 @@ steps:
...
       displayName: 'Generate "WorkloadRollback.json"'
 
-    - task: PublishPipelineArtifact@1
+    - task: 1ES.PublishPipelineArtifact@1
       displayName: 'Publish WorkloadRollback.json'
       inputs:
-        targetPath: $(Build.SourcesDirectory)/WorkloadRollback.json
-        artifactName: '${{ parameters.uploadPrefix }}WorkloadRollback'
+        path: $(Build.SourcesDirectory)/WorkloadRollback.json
+        artifact: '${{ parameters.uploadPrefix }}WorkloadRollback'
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/build/build-pkgs.yml:84]: They could have reused the same names :/ Would have made our diff much smaller.

File: tools/devops/automation/templates/release/vs-insertion-prep.yml [hunk 2]
```diff
@@ -103,13 +107,15 @@ stages:
...
-        displayName: Publish Nugets to dotnet8
+      - task: 1ES.PublishNuget@1
+        displayName: Publish Nugets package
         inputs:
-          command: push
+          useDotNetTask: false # The default is false to use the NuGetCommand task. Set to true to use the DotNetCoreCLI task to publish packages.
           packagesToPush: '$(Build.SourcesDirectory)/package/*.nupkg'
+          packageParentPath: '$(Build.SourcesDirectory)/package'
           nuGetFeedType: external
-          publishFeedCredentials: dnceng-dotnet8
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/release/vs-insertion-prep.yml:112]: Which feed do we publish to now? It doesn't look like that's chosen anywhere.

**Review by rolfbjarne — APPROVED** [review]

**Review by mcumming — APPROVED** [review]

File: tools/devops/automation/templates/build/build-pkgs.yml (hunk 1 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/build/build-pkgs.yml:84]: Trust me, I have rant about this.

File: tools/devops/automation/templates/release/vs-insertion-prep.yml (hunk 2 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/release/vs-insertion-prep.yml:112]: yeah, I requested a force insert to check that

**Review by dalexsoto — APPROVED** [review]

**Review by haritha-mohan — APPROVED** [review]

File: tools/devops/automation/templates/build/build-pkgs.yml (hunk 1 above)
Comment (mcumming) [reply tools/devops/automation/templates/build/build-pkgs.yml:84]: They are the same if you use the `templateContext.outputs`, maybe it was intentional to make you move to that format? evil geniuses?

**Pull request closed without merging**